ansible_events_ui/db/utils/lostream.py

```diff
@@ -0,0 +1,227 @@
+
+    async def write(self, buff: Union[str, bytes]) -> int:
+        self.closed_check()
+        if self.imode == self.INV_READ:
+            raise UnsupportedOperation("not writeable")
+
+        if len(buff) > 0:
+            sql = """
+select lo_put(:_oid, :_pos, :_buff) as lo_bytes
+;
+"""
+            if self.text_data and isinstance(buff, str):
+                buff = buff.encode("utf-8")
+
+            bufflen = len(buff)
+            await self.session.execute(
+                sql, {"_oid": self.oid, "_pos": self.pos, "_buff": buff}
+            )
+            self.pos += bufflen
+
+            return bufflen
+        else:
+            return 0
+
+    async def flush(self) -> None:
+        self.closed_check()
+        if self.imode == self.INV_READ:
+            raise UnsupportedOperation("not writeable")
+        await self.session.commit()
+
+    async def truncate(self) -> None:
+        self.closed_check()
+        if self.imode == self.INV_READ:
+            raise UnsupportedOperation("not writeable")
+
+        open_sql = """
+select lo_open(:_oid, :_mode) as lofd;
+        """
+        trunc_sql = """
+select lo_truncate(:_fd, :_len);
+        """
+        close_sql = """
+select lo_close(:_fd) as lofd;
+        """
+        cur = await self.session.execute(
+            open_sql, {"_oid": self.oid, "_mode": self.imode}
+        )
+        fd = cur.first().lofd
+        await self.session.execute(trunc_sql, {"_fd": fd, "_len": self.pos})
+        await self.session.execute(close_sql, {"_fd": fd})
+        self.length = self.pos
+
+    async def close(self) -> None:
+        if (
+            not self.closed
+            and self.oid > 0
+            and (self.imode & self.INV_WRITE)
+            and (self.pos != self.length)
+        ):
+            await self.truncate()
+
+        if self.imode & self.INV_WRITE:
+            await self.flush()
+
+    async def delete(self) -> None:
+        if self.oid is not None and self.oid > 0:
+            await self.session.execute(
+                """
+select lo_unlink(:oid) ;
+                """,
+                {"oid": self.oid},
+            )
+            self.oid = 0
+
+
+async def _create_large_object(session: AsyncSession) -> int:
+    sql = """
+select lo_create(0) as loid;
+    """
+    loid = (await session.execute(sql)).first().loid
+    await session.commit()
+    return loid
+
+
+async def _verify_large_object(oid: int, session: AsyncSession) -> Tuple:
+    sql = """
+select m.oid,
+       sum(length(coalesce(d.data, convert_to('', 'utf-8')))) as _length
+  from pg_largeobject_metadata m
+  left
+  join pg_largeobject d
+    on d.loid = m.oid
+ where m.oid = :_oid
+ group
+   by m.oid;
+    """
+    rec = (await session.execute(sql, {"_oid": oid})).first()
+
+    return (rec.oid, rec._length) if rec else (None, None)
+
+
+async def large_object_factory(
+    oid: int = 0,
+    mode: str = "rb",
+    session: AsyncSession = Depends(get_db_session),
```

Comment:
This function is not used as a dependency, so the `Depends` has no use here.
`Depends` is used for FastAPI dependency injection mechanism. Dependencies are resolved when used in routes or in other dependencies.
This function is not designed to be used as a dependency, so you should not use `Depends` here.
Also session should not be optional here.